flatson/flatson.py

```diff
@@ -137,3 +124,20 @@ def flatten_dict(self, obj):
         """Return an OrderedDict dict preserving order of keys in fieldnames
         """
         return OrderedDict(zip(self.fieldnames, self.flatten(obj)))
+
+    def infer_flattened_field_names(self, schema, field_sep='.'):
```

Comment:
We can also consider this other refactor (I know, I keep refactoring that method... On my defense I will say that it's asking me to be refactored). https://github.com/scrapinghub/flatson/pull/11/files#diff-c822b2a8e759d565055bd7306c3d9928R24

It's a good point that top-level keys being considerated... I agree that it should be the expected behaviour. Let's agree a refactoring approach and handle that better.